lib/ReactGridLayout.jsx

```diff
@@ -312,12 +328,27 @@ export default class ReactGridLayout extends React.Component<Props, State> {
         newLayoutBase,
         nextProps.children,
         nextProps.cols,
-        this.compactType(nextProps)
+        compactType(nextProps)
       );
-      const oldLayout = this.state.layout;
-      this.setState({ layout: newLayout });
-      this.onLayoutMaybeChanged(newLayout, oldLayout);
+
+      return {
+        layout: newLayout,
+        // We need to save these props to state for using
+        // getDerivedStateFromProps instead of componentDidMount (in which we would get extra rerender)
+        compactType: nextProps.compactType,
+        children: nextProps.children,
+        propsLayout: nextProps.layout
+      };
     }
+
+    return null;
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    const newLayout = this.state.layout;
```

Comment:
This still triggers too many times (give it a shot with a `console.log`). Do think we solve it by changing it to the following:

```
    if (!this.state.activeDrag) {
      const newLayout = this.state.layout;
      const oldLayout = prevState.layout;

      this.onLayoutMaybeChanged(newLayout, oldLayout);
    }
```

Similar to what you're doing in the `getDerivedStateFromProps`, or will this break something? I tried it locally and everything seems to be working just fine. However it should improve performance even more. What do you think?

Reply:
@n1ghtmare done

Reply:
@daynin From what I see, this change shouldn't break anything. Can you confirm this? Otherwise performance is great. @STRML Do you want to take a quick look at this, when you get some time?

Reply:
That's a nice optimization. It shouldn't break anything as the grid should only move in two circumstances:

1. An active drag, or
2. An update to the `layout` property from the parent. 

2) doesn't apply here so we should be okay :+1: